lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,380 @@
+                fontSize: 20,
+                height: 30 / 20,
+              ).merge(weightVariableTextStyle(context, wght: 600)),
+              overflow: TextOverflow.ellipsis,
+              textAlign: TextAlign.center)),
+          _buildNextButton(context),
+        ]));
+  }
+}
+
+class _NewDmSearchBar extends StatelessWidget {
+  const _NewDmSearchBar({
+    required this.searchController,
+    required this.selectedUserIds,
+  });
+
+  final TextEditingController searchController;
+  final Set<int> selectedUserIds;
+
+  Widget _buildSearchField(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final hintText = selectedUserIds.isEmpty
+      ? zulipLocalizations.newDmSheetSearchHintEmpty
+      : zulipLocalizations.newDmSheetSearchHintSomeSelected;
+
+    return TextField(
+      controller: searchController,
+      cursorColor: designVariables.foreground,
+      style: TextStyle(
+        color: designVariables.textMessage,
+        fontSize: 17,
+        height: 22 / 17),
+      scrollPadding: EdgeInsets.zero,
+      decoration: InputDecoration(
+        isDense: true,
+        contentPadding: EdgeInsets.zero,
+        hintText: hintText,
+        hintStyle: TextStyle(
+          color: designVariables.labelSearchPrompt,
+          fontSize: 17,
+          height: 22 / 17),
+        border: InputBorder.none));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    return Container(
+      constraints: const BoxConstraints(
+        minHeight: 44,
+        maxHeight: 124),
+      padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 11),
+      decoration: BoxDecoration(color: designVariables.bgSearchInput),
+      child: SingleChildScrollView(
+        reverse: true,
+        child: Row(children: [
+          Expanded(child: Wrap(
+            spacing: 6,
+            runSpacing: 4,
+            crossAxisAlignment: WrapCrossAlignment.center,
+            children: [
+              ...selectedUserIds.map((userId) => _SelectedUserChip(userId: userId)),
+              IntrinsicWidth(child: _buildSearchField(context)),
+            ])),
+        ])));
+  }
+}
+
+class _SelectedUserChip extends StatelessWidget {
+  const _SelectedUserChip({required this.userId});
+
+  final int userId;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    final user = store.getUser(userId);
+
+    return Container(
+      decoration: BoxDecoration(
+        color: designVariables.bgMenuButtonSelected,
+        borderRadius: BorderRadius.circular(3)),
+      child: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Avatar(userId: userId, size: 22, borderRadius: 3),
```

Comment:
nit: can make compact and reduce indentation:
```suggestion
      child: Row(mainAxisSize: MainAxisSize.min, children: [
        Avatar(userId: userId, size: 22, borderRadius: 3),
```